src/sources/opentelemetry/log.rs

```diff
@@ -0,0 +1,254 @@
+            pipeline: cx.out,
+            acknowledgements,
+        })
+        .accept_gzip();
+        let source =
+            run_grpc_server(self.address, tls_settings, service, cx.shutdown).map_err(|error| {
+                error!(message = "Source future failed.", %error);
+            });
+
+        Ok(Box::pin(source))
+    }
+
+    fn outputs(&self, _global_log_namespace: LogNamespace) -> Vec<Output> {
+        if self.multiple_outputs {
+            vec![Output::default(DataType::Log).with_port(LOGS)]
+        } else {
+            vec![Output::default(DataType::Log)]
+        }
+    }
+
+    fn source_type(&self) -> &'static str {
+        "opentelemetry"
+    }
+
+    fn resources(&self) -> Vec<Resource> {
+        vec![Resource::tcp(self.address)]
+    }
+
+    fn can_acknowledge(&self) -> bool {
+        true
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct Service {
+    pipeline: SourceSender,
+    acknowledgements: bool,
+}
+
+#[tonic::async_trait]
+impl LogsService for Service {
+    async fn export(
+        &self,
+        request: Request<ExportLogsServiceRequest>,
+    ) -> Result<Response<ExportLogsServiceResponse>, Status> {
+        let mut events: Vec<Event> = request
+            .into_inner()
+            .resource_logs
+            .into_iter()
+            .flat_map(|v| v.into_iter())
+            .collect();
+
+        let count = events.len();
+        let byte_size = events.size_of();
+
+        emit!(EventsReceived { count, byte_size });
+
+        let receiver = BatchNotifier::maybe_apply_to(self.acknowledgements, &mut events);
+
+        self.pipeline
+            .clone()
+            .send_batch(events)
+            .map_err(|error| {
+                let message = error.to_string();
+                emit!(StreamClosedError { error, count });
+                Status::unavailable(message)
+            })
+            .and_then(|_| handle_batch_status(receiver))
+            .await?;
+        Ok(Response::new(ExportLogsServiceResponse {}))
+    }
+}
+
+async fn handle_batch_status(receiver: Option<BatchStatusReceiver>) -> Result<(), Status> {
+    let status = match receiver {
+        Some(receiver) => receiver.await,
+        None => BatchStatus::Delivered,
+    };
+
+    match status {
+        BatchStatus::Errored => Err(Status::internal("Delivery error")),
+        BatchStatus::Rejected => Err(Status::data_loss("Delivery failed")),
+        BatchStatus::Delivered => Ok(()),
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::{
+        opentelemetry::{
+            Common::{any_value, AnyValue, KeyValue},
+            LogService::logs_service_client::LogsServiceClient,
+            Logs::{LogRecord, ResourceLogs, ScopeLogs},
+            Resource as OtelResource,
+        },
+        test_util::{
+            self,
+            components::{assert_source_compliance, SOURCE_TAGS},
+        },
+        SourceSender,
+    };
+
+    #[tokio::test]
+    #[allow(deprecated)]
```

Comment:
What is the deprecated thing we are allowing here?

Reply:
Unsure - removing it doesn't appear to impact clippy or testing so we should go ahead and drop it.
```suggestion
```